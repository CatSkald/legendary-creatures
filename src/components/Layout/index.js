import "../../styles/global.scss";
import "../../styles/theme-light.scss";
import "../../styles/theme-dark.scss";
import styles from "./index.module.scss";

import React, { useState } from "react";
import Header from "../Header";
import Footer from "../Footer";

const LocaleContext = React.createContext();

const Layout = ({ children, pageContext: { language } }) => {
  const [isDarkColorTheme, switchColorTheme] = useState(false);

  function handleToggleColorTheme() {
    switchColorTheme(!isDarkColorTheme);
  }

  return (
    <LocaleContext.Provider value={{ language }}>
      <div
        className={`body ${styles.Wrapper} ${
          isDarkColorTheme ? "theme-dark" : "theme-light"
        }`}
      >
        <Header
          isDarkColorTheme={isDarkColorTheme}
          handleToggleColorTheme={handleToggleColorTheme}
        />
        <section className={styles.SiteContent} role="main">
          <div className={styles.Container}>{children}</div>
        </section>
        <Footer />
      </div>
    </LocaleContext.Provider>
  );
};

export { Layout, LocaleContext };
