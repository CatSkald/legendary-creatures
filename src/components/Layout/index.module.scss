.layout__container {
  display: flex;
  min-height: 100vh;
  flex-direction: column;
}

.layout {
  display: flex;
  flex: 1;
  width: 100%;
}

.layout__content {
  width: 100%;
  max-width: var(--container-width);
  margin: 0 auto;
  padding: var(--space-sm);

  @media screen and (min-width: 768px /*--screen-medium*/) {
    padding: var(--space);
  }
}
