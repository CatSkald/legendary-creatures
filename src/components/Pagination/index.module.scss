.pagination {
  align-items: center;
  color: var(--content-text);
  display: flex;
  justify-content: center;
  list-style: none;

  p {
    margin: 0;
    margin-top: var(--space-sm);
    padding: 0 var(--space);
  }

  a {
    color: var(--content-link);
    text-decoration: none;
    transition: color 0.5s;

    &:hover {
      color: var(--content-link-hover);
    }
  }
}
