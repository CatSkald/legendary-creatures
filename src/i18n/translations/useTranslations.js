import React from "react";
import { useStaticQuery, graphql } from "gatsby";
import { LocaleContext } from "../../components/Layout";

function useTranslations() {
  const { language } = React.useContext(LocaleContext);
  const { rawData } = useStaticQuery(queryTranslations);

  const translationsForCurrentLocale = rawData.edges
    .filter(item => item.node.name === language.code)
    .map(item => item.node.translations)[0];

  return translationsForCurrentLocale;
}

export default useTranslations;

const queryTranslations = graphql`
  query queryTranslations {
    rawData: allFile(filter: { sourceInstanceName: { eq: "translations" } }) {
      edges {
        node {
          name
          translations: childTranslationsYaml {
            Description
            Title
            ShortTitle
            NotFound404Header
            NotFound404Content
            footerMoto
            by
            for
            fromPerson
            fromPlace
            of
            Categories
            View
            all
            orSearchBy
            tags
            All
            Home
            Latest
            Next
            Prev
            Tags
            plural
            Number
            Origin
            Habitat
            CreaturesPageTitle
            TagsPageTitle
            origin
            categories
            number
            habitat
          }
        }
      }
    }
  }
`;
