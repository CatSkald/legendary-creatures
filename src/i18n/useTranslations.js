import React from "react";
import { useStaticQuery, graphql } from "gatsby";
import { LocaleContext } from "../components/Layout";

function useTranslations() {
  const { locale } = React.useContext(LocaleContext);
  const { rawData } = useStaticQuery(queryTranslations);

  const translationsForCurrentLocale = rawData.edges
    .filter(item => item.node.name === locale)
    .map(item => item.node.translations)[0];

  return translationsForCurrentLocale;
}

export default useTranslations;

const queryTranslations = graphql`
  query queryTranslations {
    rawData: allFile(
      filter: { sourceInstanceName: { eq: "translations" } }
    ) {
      edges {
        node {
          name
          translations: childTranslationsYaml {
            Description
            Title
            NotFound404Header
            NotFound404Content
            footerMoto
            Icon
            by
            for
            fromPerson
            fromPlace
            of
            using
            Categories
            All
            Home
            Latest
            Next
            Prev
            Tags
            plural
            Number
            Origin
            Habitat
          }
        }
      }
    }
  }
`;
