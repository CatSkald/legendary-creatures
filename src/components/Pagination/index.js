import styles from "./index.module.scss";

import React from "react";
import PropTypes from "prop-types";
import LocalizedLink from "../LocalizedLink";
import useTranslations from "../../i18n/translations/useTranslations";

const Pagination = props => {
  const translations = useTranslations();

  return (
    <ul className={styles.PaginationContainer}>
      <li>
        {!props.isFirst && (
          <LocalizedLink to={props.prevPage}>
            ← {translations.Prev}
          </LocalizedLink>
        )}
      </li>
      <p>
        {props.currentPage} {translations.of} {props.numPages}
      </p>
      <li>
        {!props.isLast && (
          <LocalizedLink to={props.nextPage}>
            {translations.Next} →
          </LocalizedLink>
        )}
      </li>
    </ul>
  );
};

Pagination.propTypes = {
  isFirst: PropTypes.bool.isRequired,
  isLast: PropTypes.bool.isRequired,
  currentPage: PropTypes.number.isRequired,
  numPages: PropTypes.number.isRequired,
  prevPage: PropTypes.string,
  nextPage: PropTypes.string,
};

export default Pagination;
