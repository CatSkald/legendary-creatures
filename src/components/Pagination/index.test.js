import React from "react";
import { render } from "@testing-library/react";
import { LocaleContext } from "../Layout";

import Pagination from "../Pagination";
import useTranslations from "../../i18n/translations/useTranslations";

jest.mock("../../i18n/translations/useTranslations");

useTranslations.mockImplementation(() => ({
  Prev: "Previous",
  of: "of",
  Next: "Next",
}));

describe("Pagination", () => {
  test.each([
    [0, 1, null, null],
    [0, 2, null, "a"],
    [1, 2, "b", null],
    [1, 3, "c", "d"],
  ])("renders correctly", (currentPage, numPages, prevPage, nextPage) => {
    const language = { path: "/", code: "en" };
    const { container } = render(
      <LocaleContext.Provider value={{ language }}>
        <Pagination
          currentPage={currentPage}
          numPages={numPages}
          prevPage={prevPage}
          nextPage={nextPage}
        />
      </LocaleContext.Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
