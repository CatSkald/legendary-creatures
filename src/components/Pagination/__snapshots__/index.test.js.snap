// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders correctly 1`] = `
<div>
  <div
    class="pagination"
  >
    <p />
    <p>
      1
       
      of
       
      1
    </p>
    <p />
  </div>
  ,
</div>
`;

exports[`Pagination renders correctly 2`] = `
<div>
  <div
    class="pagination"
  >
    <p />
    <p>
      1
       
      of
       
      2
    </p>
    <p>
      <a
        href="/a"
        hreflang="en"
      >
        Next
         →
      </a>
    </p>
  </div>
  ,
</div>
`;

exports[`Pagination renders correctly 3`] = `
<div>
  <div
    class="pagination"
  >
    <p>
      <a
        href="/b"
        hreflang="en"
      >
        ← 
        Previous
      </a>
    </p>
    <p>
      2
       
      of
       
      2
    </p>
    <p />
  </div>
  ,
</div>
`;

exports[`Pagination renders correctly 4`] = `
<div>
  <div
    class="pagination"
  >
    <p>
      <a
        href="/c"
        hreflang="en"
      >
        ← 
        Previous
      </a>
    </p>
    <p>
      2
       
      of
       
      3
    </p>
    <p>
      <a
        href="/d"
        hreflang="en"
      >
        Next
         →
      </a>
    </p>
  </div>
  ,
</div>
`;
